### PR TITLE
ARM template for creating aml compute

### PR DESCRIPTION
**Pull request by ragargms**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Comment by ragargms** [comment]
@bmoore-msft for review

File: 101-machine-learning-compute-create-amlcompute/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Create an Azure Machine Learning Workspace
+
+![Azure Public Test Date](https://azurequickstartsservice.blob.core.windows.net/badges/101-machine-learning-create/PublicLastTestDate.svg)
```
Comment (bmoore-msft) [inline 101-machine-learning-compute-create-amlcompute/README.md:3]: fix all buttons/links/badges/links

File: 101-machine-learning-compute-create-amlcompute/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            }
+        },        
+        "maxNodeCount": {
+            "defaultValue": 1,
+            "type": "int",
+            "metadata": {
+                "description": " The maximum number of nodes to use on the cluster. If not specified, defaults to 4."
+            }
+        },
+        "workspacelocation": {
```
Comment (bmoore-msft) [inline 101-machine-learning-compute-create-amlcompute/azuredeploy.json:31]: change this to "location" and set the defaultValue to resourceGroup().location

File: 101-machine-learning-compute-create-amlcompute/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            }
+        },
+        "vmSize": {
+            "defaultValue": "Standard_DS3_v2",
+            "type": "string",
+            "metadata": {
+                "description": " The size of agent VMs. More details can be found here: https://aka.ms/azureml-vm-details."
+            }
+        },
+        "subnetId": {
```
Comment (bmoore-msft) [inline 101-machine-learning-compute-create-amlcompute/azuredeploy.json:56]: break this apart into multiple parameters, one for the vnet resourceGroup, vnet name, and subnet name...

File: 101-machine-learning-compute-create-amlcompute/azuredeploy.parameters.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "workspaceName": {
+      "value": "GET-PREREQ-workspaceResource"
+    },
+    "workspacelocation": {
```
Comment (bmoore-msft) [inline 101-machine-learning-compute-create-amlcompute/azuredeploy.parameters.json:8]: remove this param and use the defaultValue

File: 101-machine-learning-compute-create-amlcompute/prereqs/prereq.azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+  "parameters": {
+      "workspaceName": {
+        "type": "string",
+        "metadata": {
+          "description": "Specifies the name of the Azure Machine Learning workspace."
+        }
+      },
+      "location": {
+        "type": "string",
+        "allowedValues": [
```
Comment (bmoore-msft) [inline 101-machine-learning-compute-create-amlcompute/prereqs/prereq.azuredeploy.json:13]: remove allowedValues

File: 101-machine-learning-compute-create-amlcompute/prereqs/prereq.azuredeploy.parameters.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "workspaceName": {
+      "value": "GEN-UNIQUE"
+    },
+    "location": {
```
Comment (bmoore-msft) [inline 101-machine-learning-compute-create-amlcompute/prereqs/prereq.azuredeploy.parameters.json:8]: remove this and use a defaultValue

File: 101-machine-learning-compute-create-amlcompute/prereqs/prereq.azuredeploy.parameters.json [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+      "value": "westcentralus"
+    },
+    "sku": {
```
Comment (bmoore-msft) [inline 101-machine-learning-compute-create-amlcompute/prereqs/prereq.azuredeploy.parameters.json:11]: remove this

File: 101-machine-learning-compute-create-amlcompute/prereqs/prereq.azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+        "defaultValue": "Disabled",
+          "allowedValues": [
+            "Enabled",
+            "Disabled"
+          ],
+          "metadata": {
+            "description": "Specifies if the Azure Machine Learning workspace should be encrypted with customer managed key."
+          }
+      },
+      "cmk_keyvault":{
```
Comment (bmoore-msft) [inline 101-machine-learning-compute-create-amlcompute/prereqs/prereq.azuredeploy.json:150]: remove the parameters and resources  that are not being used

**Comment by ragargms** [comment]
@bmoore-msft appreciate your review. We have a customer waiting on this.

**Pull request merged**